Makefile.include

```diff
@@ -708,6 +708,13 @@ BUILDDEPS += pkg-prepare
 BUILDDEPS += $(APPDEPS)
 BUILDDEPS += $(MODULE_LIST_DIR)
 
+# Ugly hack for when clean is among the target list:
+# - add clean to BUILDDEPS, so that clean completes before building starts
+# - mark generated headers as phony, so that they are rebuild
+ifneq (,$(filter clean,$(MAKECMDGOALS)))
```

Comment:
this seems dangerous, as now the clean target is somewhere in the middle of the BUILDDEPS, and some of them are created inside BINDIR (which gets cleaned by `clean`).
Is this needed with the addition below?